File: frontend/src/types/FireDevice.ts
export interface FireDevice {
  id: number;
  building_id: number;
  device_code: string;
  device_type: string;
  floor: string;
  location_desc: string;
  install_date: string;
  status: string; // DeviceStatus: AVAILABLE / UNAVAILABLE
  next_maintenance_at: string;
}

File: backend/src/models/fire_device.py
from pydantic import BaseModel


class FireDevice(BaseModel):
    id: int
    building_id: int
    device_code: str
    device_type: str
    floor: str
    location_desc: str
    install_date: str
    status: str                  # AVAILABLE / UNAVAILABLE
    next_maintenance_at: str
